src/Adapter/Order/CommandHandler/DeleteCartRuleFromOrderHandler.php

```diff
@@ -69,6 +77,10 @@ public function handle(DeleteCartRuleFromOrderCommand $command)
             throw new OrderException('Invalid cart provided.');
         }
 
+        $this->contextStateManager
```

Comment:
Same here you need a `try/finally` block and call `$this->contextStateManager->restoreContext();`, and move all the following code in a function to make review easier (or else the whole file is modified for indentation)